### DISCUSSION TITLE
Your environment is not configured properly and you will not be able to execute local builds

### DISCUSSION THREAD

**Discussion by JimLynchCodes** [NativeScript/NativeScript#9970] [Q&A]
### Issue Description

I have setup my environment, but I am still getting warnings... tns will not create a build even though things are already installed!

### Reproduction

reproduced in this repo: https://github.com/JimLynchCodes/Study-App

### Relevant log output (if applicable)

```shell
when I run `ns doctor ios`


You are using the deprecated nsconfig.json file. Just be aware that NativeScript now has an improved nativescript.config.(js|ts) file for when you're ready to upgrade this project.
✔ Getting environment information 

There seem to be issues with your configuration.
✔ Getting NativeScript components versions information...
✔ Component nativescript has 8.2.3 version and is up to date.
⚠ Update available for component tns-core-modules. Your current version is 6.5.10 and the latest available version is 6.5.27.
✔ Component tns-ios has * version and is up to date.
✔ Xcode is installed and is configured properly.
✔ xcodeproj is installed and is configured properly.
✔ CocoaPods update is not required.
✔ Your current CocoaPods version is newer than 1.0.0.
✔ Xcode version 13.4.1 satisfies minimum required version 10.
✖ WARNING: CocoaPods is not installed or is not configured properly. 
 You will not be able to build your projects for iOS if they contain plugin with CocoaPod file.
To be able to build such projects, verify that you have installed CocoaPods (`sudo gem install cocoapods`). 

✖ WARNING: Couldn't retrieve installed python packages. 
 We cannot verify your python installation is setup correctly. Please, make sure you have both 'python' and 'pip' installed.
Error while validating Python packages. Error is: Command failed: python -c "import six"
python: error: Failed to locate 'python'.
xcode-select: Failed to locate 'python', requesting installation of command line developer tools.
 

✖ WARNING: The Python 'six' package not found. 
 This package is required by the Debugger library (LLDB) for iOS. You can install it by first making sure you have pip installed and then running 'pip install six' from the terminal. 

Your environment is not configured properly and you will not be able to execute local builds.
Verify that your environment is configured according to the system requirements described at
https://docs.nativescript.org/environment-setup.html#macos-ios.
```

When I run `sudo gem install cocoapods`, output says it's installed, but ns doctor still gives the error...
```
Successfully installed cocoapods-1.11.3
Parsing documentation for cocoapods-1.11.3
Done installing documentation for cocoapods after 1 seconds
1 gem installed
```

Running `python3 -m pip install six`:
```
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: six in /Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/site-packages (1.15.0)
```
```


### Environment

<!-- COPY START -->
```yaml
OS: macOS 12.4
CPU: (12) x64 Intel(R) Core(TM) i7-8700B CPU @ 3.20GHz
Shell: /bin/zsh
node: 13.13.0
npm: 6.14.4
nativescript: 8.2.3

# android
java: Not Found
ndk: Not Found
apis: Not Found
build_tools: Not Found
system_images: Not Found

# ios
xcode: 13.4.1/13F100
cocoapods: Not Found
python: Not Found
python3: 3.8.9
ruby: 2.6.8
platforms: 
  - DriverKit 21.4
  - iOS 15.5
  - macOS 12.3
  - tvOS 15.4
  - watchOS 8.5
```

### Dependencies

```json
"dependencies": {
  "@angular/animations": "~8.2.0",
  "@angular/common": "~8.2.0",
  "@angular/compiler": "~8.2.0",
  "@angular/core": "~8.2.0",
  "@angular/forms": "~8.2.0",
  "@angular/platform-browser": "~8.2.0",
  "@angular/platform-browser-dynamic": "~8.2.0",
  "@angular/router": "~8.2.0",
  "@nativescript/theme": "~2.3.0",
  "nativescript-angular": "~8.21.0",
  "nativescript-auth0": "^3.0.2",
  "nativescript-plugin-firebase": "^10.6.3",
  "nativescript-ui-sidedrawer": "~8.0.0",
  "reflect-metadata": "~0.1.12",
  "rxjs": "^6.4.0",
  "tns-core-modules": "~6.5.0",
  "zone.js": "~0.9.1"
},
"devDependencies": {
  "@angular/compiler-cli": "~8.2.0",
  "@nativescript/schematics": "^9.0.0",
  "@ngtools/webpack": "~8.2.0",
  "codelyzer": "~4.5.0",
  "nativescript-angular-cli": "^0.1.9",
  "nativescript-dev-webpack": "~1.5.0",
  "node-sass": "^4.7.1",
  "tns-ios": "6.5.3",
  "tslint": "~5.19.0",
  "typescript": "~3.5.3"
}
```

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/nativescript-cli/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by rigor789** [comment]
1. I'm guessing your gems are not in your system path: 
   1. find the pathhttps://stackoverflow.com/questions/19072070/how-to-find-where-gem-files-are-installed
   2. add it to your system path: `export PATH="<gemspath>:$PATH"` in `.zshrc` or `.bashrc` etc (depending on the shell you are using)
2. Could be related to macos 12.4, try:
   1. `brew install python` (make sure it's in the path if it's not added by default, brew would print instructions if that's the case if I remember correctly)
   2. `sudo ln -s -f $(which python3) /usr/local/bin/python`

**Comment by JimLynchCodes** [comment]
thanks @rigor789!

I checked where the cocoapods got installed by doing:
```
which gem cocoapods
```

that output this:
```
/usr/bin/gem
```

So I added that in addition to ruby path's already in my ~/.bash_profile
```
# Add rubygems to the path
export PATH=/opt/homebrew/lib/ruby/gems/2.7.0/bin:$PATH

# or
export PATH=/usr/local/lib/ruby/gems/2.7.0/bin:$PATH


export PATH=/usr/bin/gem:$PATH

```

But when I run `tns doctor ios` I still get this warning that it can't find cocoapods... 😢

**Reply by rigor789** [reply]
Run `which pod` and lmk what you get, and then also try `pod --version`.

**Reply by JimLynchCodes** [reply]
thanks @rigor789 

here's my output for the two commands

`which pod` -->  /usr/local/bin/pod

`pod --version` --> 1.11.3

**Comment by JimLynchCodes** [comment]
I am running on mac 12.4

After running those two python commands though I still get the python warning, "The Python 'six' package not found. "

**Comment by JimLynchCodes** [comment]
Also, I don't understand this... if these are just "warnings" why are they preventing the build from happening when I run `tns build ios`??

**Reply by rigor789** [reply]
CocoaPods is required, as that needs to install your 3rd party dependencies, `six` likely wouldn't affect the build command...

**Reply by jarrodwhitley** [reply]
This warning seems to imply that is does 
![image](https://github.com/NativeScript/NativeScript/assets/11711674/e887c1b2-a5bf-4e94-97c3-96b5fd285c0e)

**Comment by JimLynchCodes** [comment]
hi @rigor789 

I ran the commands and added things into my .bashrc as you suggested, but I'm still getting these two errors:

```
✖ WARNING: CocoaPods is not installed or is not configured properly. 
 You will not be able to build your projects for iOS if they contain plugin with CocoaPod file.
To be able to build such projects, verify that you have installed CocoaPods (`sudo gem install cocoapods`). 

✖ WARNING: The Python 'six' package not found. 
 This package is required by the Debugger library (LLDB) for iOS. You can install it by first making sure you have pip installed and then running 'pip install six' from the terminal. 

```

No tns command works anymore so it's kind of a big problem... 😢

**Comment by nenzan** [comment]
any update for this? I had same issue 
<img width="638" alt="Screenshot 2022-12-08 at 10 52 20" src="https://user-images.githubusercontent.com/28696989/206352479-92df45ee-956f-48cb-b9e1-d446036ab7e2.png">

**Comment by NathanWalker** [comment]
Hi @nenzan we have been revamping the environment setup and you can try the revised instructions on the beta docs here:
https://beta.docs.nativescript.org/setup/macos#installing-python-and-six

Let us know if that helps you when following verbatim.

**Reply by nenzan** [reply]
Still same nothing change, the environment setup instruction not so different than before. Still show same error, the `six` package and python environment not detected

**Reply by NathanWalker** [reply]
Thanks for confirming. When get a chance to run this command and paste output here would be very helpful:

echo 'y' | npx nativescript-envinfo

cc/ @rigor789

**Reply by nenzan** [reply]
here's my envinfo, hope this will help


### Environment
```yaml
OS: macOS 13.0.1
CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
Shell: /bin/zsh
node: 18.8.0
npm: 8.18.0
nativescript: 8.4.0

# android
java: 18.0.2
ndk: Not Found
apis: Not Found
build_tools: Not Found
system_images: Not Found

# ios
xcode: 14.1/14B47b
cocoapods: 1.11.3
python: Not Found
python3: 3.10.8
ruby: 2.7.7
platforms: 
  - DriverKit 22.1
  - iOS 16.1
  - macOS 13.0
  - tvOS 16.1
  - watchOS 9.1
```

### Dependencies

```json
"dependencies": {},
"devDependencies": {}
```

**Reply by nenzan** [reply]
I got this @NathanWalker , in my case it need downgrade to specific Nativescript version 8.3 and repeat step by step from beginning and it's work
<img width="682" alt="image" src="https://user-images.githubusercontent.com/28696989/206748633-f7a5d4ce-1c77-48e0-970a-ef65fde37b6d.png">

**Reply by DepickereSven** [reply]
I was able to fix it by running the following command:
```bash
sudo ln -s /Library/Developer/CommandLineTools/usr/bin/python3 /Library/Developer/CommandLineTools/usr/bin/python
```

Because I kept getting the pop-up to install the dev tools.
Reference: https://developer.apple.com/forums/thread/704099

**Reply by lysla** [reply]
I just bought a new M2 macbook pro and I had same issue. Took hours to find your answer, that was the solution for me; thanks a lot!

**Comment by a3pelawi** [comment]
Same problem, 
I just uninstall cocoapods that installed using gem, 
`sudo gem uninstall cocoapods`
than install it again using brew.
`brew install cocoapods` 
i'm using macbook and zsh, and I never export anything in .zshrc

- And everything work good.
